### PR TITLE
Add section on cryptographic keys.

### PR DESCRIPTION
**Pull request by msporny**
Add initial content to the data model section on cryptographic keys.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-spec/pull/31.html" title="Last updated on Oct 10, 2019, 1:16 AM UTC (8aac222)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-spec/31/0c41898...8aac222.html" title="Last updated on Oct 10, 2019, 1:16 AM UTC (8aac222)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -523,8 +523,18 @@ <h2>
 
     <section>
       <h2>
-Keys
+Cryptographic Keys
```
Comment (dlongley) [inline index.html]: We should talk about "Verification Methods" somewhere and indicate that "Cryptographic Keys" are a subset.

File: index.html [hunk 2]
```diff
@@ -523,8 +523,18 @@ <h2>
...
       </h2>
+      <p>
+A DID Document can contain cryptographic keys and other material that can be
```
Comment (dlongley) [inline index.html]:
```suggestion
A DID Document can express cryptographic keys and other material that can be
```

File: index.html [hunk 3]
```diff
@@ -523,8 +523,18 @@ <h2>
...
+used to authenticate or authorize interactions with the subject of the DID or
+associated parties. These keys often contain identifiers that allow them to be
```
Comment (dlongley) [inline index.html]:
```suggestion
associated parties. The information expressed about keys often includes globally unambiguous identifiers and public key material which can be used to verify digital signatures.
```

Above suggestion might need to be manually added -- github isn't friendly with EOLs yet.

File: index.html [hunk 4]
```diff
@@ -523,8 +523,18 @@ <h2>
...
+referenced in a global network. They also contain information such as the public
+parameters for the key, which are often used to verify digital signatures. Other
+information can include status information for the key, such as if it has been
```
Comment (dlongley) [inline index.html]:
```suggestion
information can be expressed including status information for the key, such as if it has been
```

File: index.html [hunk 5]
```diff
@@ -523,8 +523,18 @@ <h2>
...
+suspended or revoked, or other information that enables one to verify if it is a
```
Comment (dlongley) [inline index.html]:
```suggestion
suspended or revoked, or attributes that enable one to verify if it is a
```

File: index.html (hunk 1 above)
Comment (msporny) [reply index.html]: Changed section header to "Verification Methods" in 6d875bf5a55ff729700ec4f47a7301ecb1ad060c.

File: index.html (hunk 2 above)
Comment (msporny) [reply index.html]: Fixed.

File: index.html (hunk 1 above)
Comment (msporny) [reply index.html]: Changed it back... the concept is so general that it's difficult to get across. Agree that we need it in there, but this PR is probably the wrong place to do it. Having a section called "Verification Methods" is confusing... "Cryptographic Keys" is an easier concept for people to grasp. Attempted to mention that "Cryptographic Keys" are one type of verification method... we need to provide other examples.

File: index.html (hunk 3 above)
Comment (msporny) [reply index.html]: Fixed.

File: index.html (hunk 4 above)
Comment (msporny) [reply index.html]: Fixed.

File: index.html (hunk 5 above)
Comment (msporny) [reply index.html]: Fixed.

File: index.html (hunk 1 above)
Comment (dmitrizagidulin) [reply index.html]: @msporny @dlongley Maybe the section header can express this tension, and be "Cryptographic Keys and Other Verification Methods" ?

File: index.html [hunk 6]
```diff
@@ -529,15 +529,26 @@ <h2>
...
-has an identifier (<code>id</code>) of its own, a <code>type</code>,
-and a <code>controller</code>, as well as other properties which
-depend on what type of key it is.
+A DID Document can express cryptographic keys and other verification methods
+that can be used to authenticate or authorize interactions with the subject of
+the DID or associated parties. The information expressed often includes globally
+unambiguous identifiers and public key material which can be used to verify
+digital signatures. Other information can be expressed such as status
+information for the key, such as if it has been suspended or revoked, or other
+attributes that enable one to verify if it is a hardware-backed cryptographic
```
Comment (dmitrizagidulin) [inline index.html]:
```suggestion
attributes that enable one to verify that it is a hardware-backed cryptographic
```

Comment (dmitrizagidulin) [reply index.html]: (More pedantically, it should be either "determine if" or "verify that")

Comment (TallTed) [reply index.html]: "determine whether" ;-)

Comment (msporny) [reply index.html]: Applied @TallTed's language.

File: index.html (hunk 1 above)
Comment (msporny) [reply index.html]: Applied @dmitrizagidulin's title in 8aac222218dbe5114b5a41a03864c520a38693a0.

**Comment by msporny** [comment]
Multiple reviews, applied suggestions by reviewers, no objections, merging.

**Pull request merged**